### PR TITLE
DiffuseProbeGridDownsample disable

### PR DESCRIPTION
**Pull request by tjmgd**
The DiffuseProbeGridDownsample pass process currently runs even when there is no Diffuse Probe Grid in a level. This can be a significant overhead for some devices. This fix prevents the problem by adding a corresponding class for the pass, and implementing an IsEnabled check.

### CODE REVIEW

**Review by Adi-Amazon — CHANGES_REQUESTED** [review]
Since the sole purpose of this class is to implement the Enable/Disable to remove the computation, this needs to be mentioned in the header so that in the future it might be changed / removed if there are other options (for example - removing the pass from a parent class pass via code might also be a viable option)

File: Gems/Atom/Feature/Common/Code/Source/DiffuseGlobalIllumination/DiffuseProbeGridDownsamplePass.cpp
Comment (Adi-Amazon) [inline Gems/Atom/Feature/Common/Code/Source/DiffuseGlobalIllumination/DiffuseProbeGridDownsamplePass.cpp:11]: nit: remove

File: Gems/Atom/Feature/Common/Code/atom_feature_common_files.cmake
Comment (Adi-Amazon) [inline Gems/Atom/Feature/Common/Code/atom_feature_common_files.cmake:135]: nit: indent

**Review by Adi-Amazon — APPROVED** [review]

**Comment by rgba16f** [comment]
Previous build failed. Started a new one today after the updates.
https://jenkins.build.o3de.org/job/O3DE/view/change-requests/job/PR-5518/

**Comment by rgba16f** [comment]
@tjmgd It looks like you have some tab characters in the file

VALIDATION FAILURE SUMMARY
VALIDATOR_FAILED: TabsValidator D:\workspace\o3de\Gems\Atom\Feature\Common\Code\atom_feature_common_files.cmake::TabsValidator FAILED TabsValidator - 1 tabs in this file
    First instance of a tab: 
         +    Source/DiffuseGlobalIllumination/DiffuseProbeGridRenderPass.h
    ---> +\tSource/DiffuseGlobalIllumination/DiffuseProbeGridDownsamplePass.cpp

**Comment by hultonha** [comment]
Kicked off another build - https://jenkins.build.o3de.org/job/O3DE/view/change-requests/job/PR-5518/

**Comment by lawsonamzn** [comment]
what happened here? was this dropped?  Looks like the build succeeded but it was 4 days ago, so another might need to be done?

**Comment by hultonha** [comment]
> what happened here? was this dropped? Looks like the build succeeded but it was 4 days ago, so another might need to be done?

This should be okay to merge, might want to update to be safe, sorry I missed hitting the merge button soon (maybe @rgba16f or @Adi-Amazon can do the honors as-is?) Seems unlikely this will cause a failure but it's a small risk

**Comment by moudgils** [comment]
> > what happened here? was this dropped? Looks like the build succeeded but it was 4 days ago, so another might need to be done?
> 
> This should be okay to merge, might want to update to be safe, sorry I missed hitting the merge button soon (maybe @rgba16f or @Adi-Amazon can do the honors as-is?) Seems unlikely this will cause a failure but it's a small risk

There was an issue with this PR. @dmcdiarmid-ly should make the final call if the PR is look too merge.

**Comment by tjmgd** [comment]
> > > what happened here? was this dropped? Looks like the build succeeded but it was 4 days ago, so another might need to be done?
> > 
> > 
> > This should be okay to merge, might want to update to be safe, sorry I missed hitting the merge button soon (maybe @rgba16f or @Adi-Amazon can do the honors as-is?) Seems unlikely this will cause a failure but it's a small risk
> 
> There was an issue with this PR. @dmcdiarmid-ly should make the final call if the PR is ok to merge.

What was the issue in question? This PR has a sister PR that is very similar so it is important to know for future reference. Can I expect a decision in the near future on this?

**Comment by moudgils** [comment]
> > > > what happened here? was this dropped? Looks like the build succeeded but it was 4 days ago, so another might need to be done?
> > > 
> > > 
> > > This should be okay to merge, might want to update to be safe, sorry I missed hitting the merge button soon (maybe @rgba16f or @Adi-Amazon can do the honors as-is?) Seems unlikely this will cause a failure but it's a small risk
> > 
> > 
> > There was an issue with this PR. @dmcdiarmid-ly should make the final call if the PR is ok to merge.
> 
> What was the issue in question? This PR has a sister PR that is very similar so it is important to know for future reference. Can I expect a decision in the near future on this?

I have pinged @dmcdiarmid-ly about this PR.

**Comment by wintermute-motherbrain** [comment]
We have a PR incoming that will address this.  Please hold off on merging this as the new PR handles edge cases not covered by this PR

**Comment by tjmgd** [comment]
Can I get a link for the new PR? When is it expected to be approved?

**Comment by wintermute-motherbrain** [comment]
Hi TJ, yes apologies I thought the PR was already ready (hopefully will be soon) and seems I was premature in closing this ticket.  Once the PR is up @dmcdiarmid-ly can link it here.  The main concern was around potential undefined behavior.

**Pull request closed without merging**
